core/src/data/networkDataSource.cpp

```diff
@@ -89,7 +91,10 @@ void NetworkDataSource::removePending(const TileID& _tileId) {
 
 void NetworkDataSource::cancelLoadingTile(const TileID& _tileId) {
     removePending(_tileId);
-    m_platform->cancelUrlRequest(constructURL(_tileId));
+    // cancel all possible requests for this tile
```

Comment:
Good catch - this is one more limitation of the current URL request system and I'll be thrilled to update it.